### PR TITLE
Improve thread name and start tests

### PR DESCRIPTION
**Pull request by kouvel**
Checking https://github.com/dotnet/coreclr/issues/16449 in CI

### CODE REVIEW

File: src/System.Threading.Thread/tests/ThreadTests.cs [hunk 1]
```diff
@@ -506,12 +506,34 @@ public static void ManagedThreadIdTest()
         [Fact]
         public static void NameTest()
         {
-            var t = new Thread(() => { });
+            string name = "a";
```
Comment (stephentoub) [inline src/System.Threading.Thread/tests/ThreadTests.cs]: Nit: to help avoid any issues, could we make this random? e.g. `string name = Guid.NewGuid().ToString("N");`?

File: src/System.Threading.Thread/tests/ThreadTests.cs [hunk 2]
```diff
@@ -506,12 +506,34 @@ public static void ManagedThreadIdTest()
...
+                    Assert.Throws<InvalidOperationException>(() => ct.Name = name + "b");
+                    Assert.Equal(name, ct.Name);
+                });
+            t.IsBackground = true;
             Assert.Null(t.Name);
-            t.Name = "a";
-            Assert.Equal("a", t.Name);
-            Assert.Throws<InvalidOperationException>(() => t.Name = "b");
-            Assert.Equal("a", t.Name);
+            t.Name = name;
```
Comment (stephentoub) [inline src/System.Threading.Thread/tests/ThreadTests.cs]: What happens if you explicitly set it to null?  Are you able to do so any number of times?

Comment (kouvel) [reply src/System.Threading.Thread/tests/ThreadTests.cs]: Yes but only if the name was not set yet, I'll add some more asserts

File: src/System.Threading.Thread/tests/ThreadTests.cs (hunk 1 above)
Comment (kouvel) [reply src/System.Threading.Thread/tests/ThreadTests.cs]: Sure will do

**Pull request merged**